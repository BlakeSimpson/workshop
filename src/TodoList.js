/**
 * This component should render the TodoItems inside a <li>
 */
import React, { Component } from 'react';
