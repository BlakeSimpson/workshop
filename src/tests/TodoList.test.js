import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoList from '../TodoList';

const items = [{
  text: 'Get to the workshop 🚶‍♀️',
  checked: true,
}, {
  text: 'Learn the basics of React 👩‍🎓',
  checked: false,
}, {
  text: 'Write some apps 💻',
  checked: false,
}];

it('should render the passed-in items', () => {
	const tree = toJson(shallow(<TodoList items={items} />));
	expect(tree).toMatchSnapshot();
});

it('should set the value of the TodoInput from the state', () => {
	const component = shallow(<TodoList items={items} />);
  component.setState({
    value: 'Test item'
  });
	expect(toJson(component)).toMatchSnapshot();
});

it('should change the value of the TodoInput from the TodoInput', () => {
	const component = shallow(<TodoList items={items} />);
  component.find('TodoInput').simulate('change', { target: { value: 'Changed input' }});
	expect(toJson(component)).toMatchSnapshot();
});

it('should add an item based on the value in the state', () => {
	const component = shallow(<TodoList items={items} />);
  const preventDefault = jest.fn();
  component.setState({
    value: 'Test item'
  });
  component.find('form').simulate('submit', { preventDefault });
	expect(toJson(component)).toMatchSnapshot();
  expect(preventDefault).toBeCalled();
});
