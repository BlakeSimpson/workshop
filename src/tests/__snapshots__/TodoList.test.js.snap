exports[`test should add an item based on the value in the state 1`] = `
<div>
  <ul>
    <li>
      <TodoItem
        checked={true}
        onClick={[Function]}
        text="Get to the workshop 🚶‍♀️" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Learn the basics of React 👩‍🎓" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Write some apps 💻" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Test item" />
    </li>
  </ul>
  <form
    onSubmit={[Function]}>
    <TodoInput
      onChange={[Function]}
      value="" />
  </form>
</div>
`;

exports[`test should change the value of the TodoInput from the TodoInput 1`] = `
<div>
  <ul>
    <li>
      <TodoItem
        checked={true}
        onClick={[Function]}
        text="Get to the workshop 🚶‍♀️" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Learn the basics of React 👩‍🎓" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Write some apps 💻" />
    </li>
  </ul>
  <form
    onSubmit={[Function]}>
    <TodoInput
      onChange={[Function]}
      value="Changed input" />
  </form>
</div>
`;

exports[`test should render the passed-in items 1`] = `
<div>
  <ul>
    <li>
      <TodoItem
        checked={true}
        onClick={[Function]}
        text="Get to the workshop 🚶‍♀️" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Learn the basics of React 👩‍🎓" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Write some apps 💻" />
    </li>
  </ul>
  <form
    onSubmit={[Function]}>
    <TodoInput
      onChange={[Function]}
      value="" />
  </form>
</div>
`;

exports[`test should set the value of the TodoInput from the state 1`] = `
<div>
  <ul>
    <li>
      <TodoItem
        checked={true}
        onClick={[Function]}
        text="Get to the workshop 🚶‍♀️" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Learn the basics of React 👩‍🎓" />
    </li>
    <li>
      <TodoItem
        checked={false}
        onClick={[Function]}
        text="Write some apps 💻" />
    </li>
  </ul>
  <form
    onSubmit={[Function]}>
    <TodoInput
      onChange={[Function]}
      value="Test item" />
  </form>
</div>
`;
